homeassistant/components/tmdb/sensor.py

```diff
@@ -0,0 +1,123 @@
+"""Support for The Movie Database API."""
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_API_KEY, ATTR_ID, ATTR_ATTRIBUTION)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ['tmdbsimple==2.2.0']
+
+DOMAIN = 'tmdb'
+
+ATTRIBUTION = "Data provided by themoviedb.org"
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_MOVIE_LISTS = 'movie_lists'
+CONF_TELEVISION_LISTS = 'television_lists'
+
+ATTR_MEDIA = 'media'
+ATTR_RELEASE_DATE = 'release_date'
+ATTR_OVERVIEW = 'overview'
+ATTR_POSTER_PATH = 'poster_path'
+ATTR_TITLE = 'title'
+
+LIST_TYPES = ['upcoming', 'now_playing', 'popular', 'top_rated']
+
+SCAN_INTERVAL = timedelta(seconds=300)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_API_KEY): cv.string,
+    vol.Optional(CONF_MOVIE_LISTS):
+        vol.All(cv.ensure_list, [vol.In(LIST_TYPES)]),
+    vol.Optional(CONF_TELEVISION_LISTS):
+        vol.All(cv.ensure_list, [vol.In(LIST_TYPES)]),
+})
+
+
+def normalize_result(result):
+    """Normalize a TMDB result."""
+    return {
+        ATTR_ID: result.get(ATTR_ID),
+        ATTR_RELEASE_DATE: result.get(ATTR_RELEASE_DATE),
+        ATTR_OVERVIEW: result.get(ATTR_OVERVIEW),
+        ATTR_POSTER_PATH: result.get(ATTR_POSTER_PATH),
+        ATTR_TITLE: result.get(ATTR_TITLE)
+    }
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up The Movie Database sensor platform."""
+    import tmdbsimple as tmdb
+    tmdb.API_KEY = config[CONF_API_KEY]
```

Comment:
What happens if the api key is invalid? If possible we should check that before adding any sensors.

Reply:
You can't validate the api key until you make the actual api call. This is already catched. I was asked this already,

Reply:
Yes, so then we should try and make and api call in `setup_platform` and if that fails, log error and return.

Reply:
make a get request just to validate the api? nah.

Reply:
This is standard for all home assistant platforms that use credentials.